osu.Game/Screens/Play/HUD/PerformancePointsCounter.cs

```diff
@@ -60,15 +60,19 @@ public PerformancePointsCounter()
             Current.Value = DisplayedCount = 0;
         }
 
+        private Mod[] clonedMods;
+
         [BackgroundDependencyLoader]
         private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
         {
             Colour = colours.BlueLighter;
 
             if (gameplayState != null)
             {
+                clonedMods = gameplayState.Mods.Select(m => m.DeepClone()).ToArray();
+
                 var gameplayWorkingBeatmap = new GameplayWorkingBeatmap(gameplayState.Beatmap);
-                difficultyCache.GetTimedDifficultyAttributesAsync(gameplayWorkingBeatmap, gameplayState.Ruleset, gameplayState.Mods.ToArray(), loadCancellationSource.Token)
+                difficultyCache.GetTimedDifficultyAttributesAsync(gameplayWorkingBeatmap, gameplayState.Ruleset, gameplayState.Mods.Select(m => m.DeepClone()).ToArray(), loadCancellationSource.Token)
```

Comment:
Any reason for not using the cloned instances here?

Reply:
I was thinking it should be sufficient to do the mod/score clone once in BDL rather than every judgement.

Reply:
Oh, I guess score does need to be cloned every time if a perfcalc instance uses it for something... Never mind